packages/utils/crates/utils-types/src/primitives.rs

```diff
@@ -56,11 +59,17 @@ impl Address {
 
     /// Creates a random Ethereum address, mostly used for testing
     pub fn random() -> Self {
+        // Uses getrandom, because it cannot bring in dependency on core-crypto
         let mut addr = [0u8; Self::SIZE];
         getrandom(&mut addr[..]).unwrap();
 
         Self { addr }
     }
+
+    /// Checks if the address is all zeroes.
+    pub fn is_zero(&self) -> bool {
+        self.addr == [0u8; Self::SIZE]
```

Comment:
This creates an array on stack and then compares the values one by one. Size check and flattening could be better.